ulib/axlibc/src/rand.rs

```diff
@@ -6,34 +6,126 @@
  *   THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT, MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
  *   See the Mulan PSL v2 for more details.
  */
+use core::ffi::{c_int, c_long, c_uint};
 
-//! Random number generator.
+use core::sync::atomic::{AtomicU64, Ordering::SeqCst};
 
-use core::{
-    ffi::{c_int, c_long, c_uint},
-    sync::atomic::{AtomicU64, Ordering::SeqCst},
-};
+static SEED: AtomicU64 = AtomicU64::new(1);
 
-static SEED: AtomicU64 = AtomicU64::new(0xa2ce_a2ce);
+/// Returns a 32-bit unsigned pseudo random interger from LCG.
+fn random_lcg_32() -> u32 {
+    let new_seed = SEED.load(SeqCst).wrapping_mul(25214903917).wrapping_add(11);
+    SEED.store(new_seed, SeqCst);
+    (new_seed >> 33) as u32
+}
+
+/// Returns a 64-bit unsigned pseudo random interger from LCG.
+fn random_lcg_64() -> u64 {
+    let new_seed = SEED.load(SeqCst).wrapping_mul(6364136223846793005) + 1442695040888963407;
+    SEED.store(new_seed, SeqCst);
+    new_seed as u64
+}
+
+/// Sets the seed for the random number generator implemented by LCG.
+fn srand_lcg(seed: u64) {
+    SEED.store(seed, SeqCst);
+}
+
+/// Checking if the CPU core is compatible with hardware random number instructions.
+#[cfg(feature = "random-hw")]
+fn has_rdrand() -> bool {
+    #[cfg(target_arch = "x86_64")]
+    {
+        let mut ecx: u32;
+        unsafe {
+            core::arch::asm!(
+                "mov eax, 1",
+                "cpuid",
+                out("ecx") ecx
+            )
+        }
+        ecx & (1 << 30) != 0
+    }
+    #[cfg(target_arch = "aarch64")]
+    {
+        let mut ID_AA64ISAR0_EL1: u64;
+        unsafe {
+            core::arch::asm!(
+                "mrs {},ID_AA64ISAR0_EL1",
+                out(reg) ID_AA64ISAR0_EL1
+            )
+        }
+        ID_AA64ISAR0_EL1 & (0b1111 << 60) == 0b0001 << 60
+    }
+    #[cfg(target_arch = "riscv64")]
+    {
+        false
+    }
+}
 
-/// Sets the seed for the random number generator.
+/// Return 64-bit unsigned random interger using cpu instruction
+#[cfg(feature = "random-hw")]
+fn random_hw() -> u64 {
+    let mut rand: u64;
+
+    #[cfg(target_arch = "x86_64")]
+    unsafe {
+        core::arch::asm! {
+            "rdrand {0:r}",
+            out(reg) rand
+        }
+    }
+
+    #[cfg(target_arch = "aarch64")]
+    unsafe {
+        core::arch::asm! {
+            "mrs {}, s3_3_c2_c4_0", // s3_3_c2_c4_0 is register `rndr`
+            out(reg) rand
+        }
+    }
+
+    #[cfg(target_arch = "riscv64")]
+    {
+        panic!("riscv64 has no rdrand instructions")
+    }
+
+    rand as u64
```

Comment:
`rand` instead of `rand as u64`

Reply:
The warning has been fixed in the lastest commit.